taichi/program/state_flow_graph.cpp

```diff
@@ -892,8 +927,33 @@ void StateFlowGraph::topo_sort_nodes() {
           previous_size, nodes_.size());
     }
   }
+  bool changed = false;
+  for (int i = 1; i < (int)nodes_.size(); i++)
+    if (nodes_[i]->node_id != i) {
+      changed = true;
+      break;
+    }
   reid_nodes();
   reid_pending_nodes();
+
+  //    if (changed) {
+  //      std::cout << "changed after topo_sort:" << std::endl;
+  //      print();
+  //
+  //      static bool first = true;
+  //      if (first) {
+  //        first = false;
+  //        auto dot = dump_dot(/*rankdir=*/std::nullopt);
+  //        auto dot_fn = std::string("test_good");
+  //        {
+  //          std::ofstream dot_file(dot_fn + ".dot");
+  //          dot_file << dot;
+  //        }
+  //        std::system(
+  //            fmt::format("dot -Tpdf -o {}.pdf {}.dot", dot_fn,
+  //            dot_fn).c_str());
+  //      }
+  //    }
```

Comment:
Remove the comments here or wrap it with an `if (false)`?

Reply:
Oh, I forgot to remove this. Good catch.